### PR TITLE
Upgrade Rocket to 0.5.0-rc.1

### PR DESCRIPTION
**Pull request by schrieveslaach**

### CODE REVIEW

**Comment by jpommerening** [comment]
FYI, Shiplift updates are on the way:
- https://github.com/softprops/shiplift/pull/280
- https://github.com/softprops/shiplift/pull/281

**Review by binepedia — APPROVED** [review]
Looks fine by me, but I'm not a big expert in Rust, so not completly sure.

File: api/src/models/app_status_change_id.rs [hunk 1]
```diff
@@ -54,8 +54,9 @@ impl std::str::FromStr for AppStatusChangeId {
 impl<'r> FromParam<'r> for AppStatusChangeId {
     type Error = AppStatusChangeIdError;
 
-    fn from_param(param: &'r RawStr) -> Result<Self, Self::Error> {
-        AppStatusChangeId::from_str(&param.url_decode()?)
+    fn from_param(param: &'r str) -> Result<Self, Self::Error> {
+        // TODO: decode?
```
[the remark this answers is not in the document]
Comment (jpommerening) [reply api/src/models/app_status_change_id.rs:58]: Looks like it is not necessary anymore :D 
From the [changelog](https://github.com/SergioBenitez/Rocket/blob/master/CHANGELOG.md#routing-and-uris):
> Percent-decoding is performed before calling FromParam implementations.

**Comment by jpommerening** [comment]
:heart: the new `async` API methods!

Didn't do a deep review or anything. Just curious :)

File: api/src/models/app_status_change_id.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (schrieveslaach) [reply api/src/models/app_status_change_id.rs:58]: Yes, that's a relic. 😅 Nice catch.

**Comment by schrieveslaach** [comment]
@jpommerening, thanks for pointing out that your PRs got merged. I switch to the upstream version of shiplift.

**Pull request merged**